pkg/logql/lex.go

```diff
@@ -240,21 +240,27 @@ func tryScanDuration(number string, l *scanner.Scanner) (time.Duration, bool) {
 		return 0, false
 	}
 	// we've found more characters before a whitespace or the end
-	d, err := time.ParseDuration(sb.String())
+	var duration time.Duration
+	prometheusDuration, err := model.ParseDuration(sb.String())
 	if err != nil {
-		return 0, false
+		duration, err = time.ParseDuration(sb.String())
```

Comment:
I think calling `sb.String()` here will have some interesting side-effects; don't we want to try parse the _same_ token twice?

Reply:
I'm updating this, but to ensure that I follow your reasoning, do you mind explaining which side-effects you anticipate?
Since we aren't writing to the string builder, I would expect calls to String() to be idempotent?

Reply:
Apologies; you're right. I misread; I thought this was reading from the scanner